static/js/copy_and_paste.js

```diff
@@ -95,6 +95,21 @@ function construct_copy_div(div, start_id, end_id) {
     }
 }
 
+// The text selection has timestamps (and also names on Chrome).
+// This function removes the names, timestamps, extra spaces and
+// the "EDITED" tags from copied text.
+function remove_names_and_times(div, selected_text) {
+    var selected_message_content = "<p>" +
+        selected_text.replace(
+            /[\n\r]((\s+)([A-Z][a-z]+|[a-z]+)(\s+)((?:(?:[0-1][0-9])|(?:[2][0-3])|(?:[0-9])):(?:[0-5][0-9])?(?:(\s+)?(?:am|AM|pm|PM))?)|(\s+)((?:(?:[0-1][0-9])|(?:[2][0-3])|(?:[0-9])):(?:[0-5][0-9])?(?:(\s+)?(?:am|AM|pm|PM))?))/g,
```

Comment:
@showell @timabbott I had to add this regex to remove Time stamps (and in the case of Chrome, user names) from the selected text. It is significantly long and I would really love some feedback on this.

Reply:
I guess the thing I was imagining was:
* Selecting with the mouse would just select text (possibly we would add noselect to the other elements?), and Ctrl+C would use the browser copy.
* The other buttons would construct the copied elements in javascript

Using a regex to remove timestamps and stuff is too brittle (and e.g. the am/pm stuff is not going to work with internationalization).

Reply:
I am not sure I understand your first point. When talking of `Ctrl + C` do you mean we should change our current `Ctrl+C` behavior to not select full messages, but rather just the selected text? 

I was definitely worried about the regex and there could still be edge cases where it may not give us the desired result. 
But, `no-select` doesn't seem to work for Chrome (although the same thing works well on Firefox). I found a [bug report](https://bugs.chromium.org/p/chromium/issues/detail?id=776233&q=user-select&colspec=ID%20Pri%20M%20Stars%20ReleaseBlock%20Component%20Status%20Owner%20Summary%20OS%20Modified) for the Chrome Browser and I think this may not have been fixed yet. 

What this leads to is a time stamp and Name (if selected) in Chrome:
![Screenshot_1](https://user-images.githubusercontent.com/25329595/55437744-4701f000-55bd-11e9-816d-62ee5eb7c424.png)

On Firefox though, it works as expected.

Reply:
I guess there was some question on what the consensus is on this decision, but my proposal here was that yeah, we should change our current Ctrl+c behavior to not select full messages. 

Are the commits separated enough that we could merge the new menu items without changing how ctrl+c works?

Reply:
No, I rebased my branch off of @vrongmeal 's changes to the Ctrl+C behavior. There was no other way to get the desired functions for the selection menu without changing the default copy behavior. We should have a concrete discussion with Tim regarding what we want with Ctrl+C.